llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -869,8 +871,10 @@ SDValue RISCVTargetLowering::lowerRETURNADDR(SDValue Op,
 
   // Return the value of the return address register, marking it an implicit
   // live-in.
-  Register Reg = MF.addLiveIn(RI.getRARegister(), getRegClassFor(XLenVT));
-  return DAG.getCopyFromReg(DAG.getEntryNode(), DL, Reg, XLenVT);
+  MCPhysReg PhysReg = RISCVABI::isCheriPureCapABI(Subtarget.getTargetABI())
```

Comment:
Should we add a .isCheriPurecap() function to subtarget?

Reply:
Maybe, though a lot of times the ABI is already lying around in order to check for soft-float/hard-single/hard-double. The subtarget currently only has helper method for querying the various ISA features, and just has a single one for getting the ABI, putting the responsibility on the caller, so I figured this was more in keeping with the current style.